Update PULL_REQUEST_TEMPLATE.md (#12313)
aptos-labs/aptos-core: .github/PULL_REQUEST_TEMPLATE.md

## Description
Please include a summary of the change, including which issue it fixes or what feature it adds. Include relevant motivation, context and documentation as appropriate. List dependencies that are required for this change, if any.

## Type of Change
- [ ] Bug fix

> - [ ] New feature

Reviewer: Some more options make it more flexible
```suggestion
- [ ] New feature
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Testing
```

- [ ] Documentation update

## Which Component or System Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

> 

Reviewer: ```suggestion

Other:

```

Author: I already have Other

## Description
Please include a summary of the change, including which issue it fixes or what feature it adds. Include relevant motivation, context and documentation as appropriate. List dependencies that are required for this change, if any.

## Type of Change

> - [ ] Bug fix
> - [ ] New feature
> - [ ] Performance improvement
> - [ ] Refactoring
> - [ ] Dependency update
> - [ ] Documentation update

Author: ```suggestion
- [ ] New feature
- [ ] Bug fix
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
```
```suggestion
- [ ] Bug fix
- [ ] New feature
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
```

## Which Component or System Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?
Please ensure that the functionality introduced by this change is well tested and verified to work as expected. Ensure tests cover both happy and unhappy paths.

- [ ] Test A
- [ ] Test B

## Key Areas to Review
- Identify any critical parts of the code that require special attention or understanding. Explain why these parts are crucial to the functionality or architecture of the project.
- Point out any areas where complex logic has been implemented. Provide a brief explanation of the logic and your approach to make it easier for reviewers to follow.

> - Highlight any areas where you are particularly concerned or unsure about the code's impact on the project. This can include potential performance or security issues, or compatibility with existing features.

Author: ```suggestion
- Highlight any areas where you are particularly concerned or unsure about the code's impact on the change. This can include potential performance or security issues, or compatibility with existing features.
```

> ## Checklist:

Author: ```suggestion
## Checklist
```

- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

> Thank you for your contribution!

Reviewer 2: This should be a comment at best, as this will be perspective of the author itself.

Author: technically a lot of the content should be a comment, but I didn't like adding `<!-- -->` everywhere. no strong feelings though

Author: added comments

- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?
<!-- Please ensure that the functionality introduced by this change is well tested and verified to work as expected. Ensure tests cover both happy and unhappy paths. -->

- [ ] Test A
- [ ] Test B

## Key Areas to Review
<!--
- Identify any critical parts of the code that require special attention or understanding. Explain why these parts are crucial to the functionality or architecture of the project.
- Point out any areas where complex logic has been implemented. Provide a brief explanation of the logic and your approach to make it easier for reviewers to follow.
- Highlight any areas where you are particularly concerned or unsure about the code's impact on the change. This can include potential performance or security issues, or compatibility with existing features.
-->

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code

> - [ ] I have commented my code, particularly in hard-to-understand areas

Author: ```suggestion
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
```

## Description
<!-- Please include a summary of the change, including which issue it fixes or what feature it adds. Include relevant motivation, context and documentation as appropriate. List dependencies that are required for this change, if any. -->

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update

## Which Component or System Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?
<!-- Please ensure that the functionality introduced by this change is well tested and verified to work as expected. Ensure tests cover both happy and unhappy paths. -->

- [ ] Test A

> - [ ] Test B

Reviewer 3: can we remove the Test A + B example? Feels kinda something that I would constantly need to delete.

> <!-- Please ensure that the functionality introduced by this change is well tested and verified to work as expected. Ensure tests cover both happy and unhappy paths. -->
> 
> - [ ] Test A
> - [ ] Test B
> 

Author: ```suggestion
<!--
- Please ensure that the functionality introduced by this change is well tested and verified to work as expected.
- Ensure tests cover both happy and unhappy paths.
- List relevant tests.
-->
```

## Description
<!-- Please include a summary of the change, including which issue it fixes or what feature it adds. Include relevant motivation, context and documentation as appropriate. List dependencies that are required for this change, if any. -->

## Type of Change
- [ ] New feature

> - [ ] Bug fix

Author: ```suggestion
- [ ] Bug fix
- [ ] Breaking change
```

- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update

> ## Which Component or System Does This Change Impact?

Reviewer 4: Can we add an option to 1) indicate if the change is a breaking change and 2) specify where/what does it?

Author: 1. added `Breaking change` in the paragraph above.
2. would that go into the `Description`?

Reviewer 4: 1. LGTM
2. I just realized where/what the break change breaks is covered in this section. So, no need to add to `Description`.

- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?
<!--
- Please ensure that the functionality introduced by this change is well tested and verified to work as expected.
- Ensure tests cover both happy and unhappy paths.

> - List relevant tests.

Author: ```suggestion
- List and link relevant tests.
```

## Description
<!-- Please include a summary of the change, including which issue it fixes or what feature it adds. Include relevant motivation, context and documentation as appropriate. List dependencies that are required for this change, if any. -->

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update

> ## Which Component or System Does This Change Impact?

Author: ```suggestion
## Which Components or Systems Does This Change Impact?
```
